src/directLinearSolvers.jl

```diff
@@ -0,0 +1,171 @@
+# implementation of direct linear solves and inversion (for symmetric p.d. matrices at this point)
+"""
+durbin(r::AbstractVector)
+computes `y = T \\ (-r)` where T = SymmetricToeplitz(vcat(1, r[1:end-1])).
+`T` is assumed to be positive definite. `r` is of length n.
+"""
+durbin(r::AbstractVector) = durbin!(zero(r), r)
+
+"""
+```
+    durbin!(y::AbstractVector, r::AbstractVector)
+```
+Same as `durbin` but uses pre-allocated vector `y` to store the result.
+"""
+function durbin!(y::AbstractVector, r::AbstractVector)
+    n = length(r)
+    length(y) == n || throw(DimensionMismatch("length(y) = $(length(y)) ≠ $n = length(r)"))
+    y[1] = -r[1]
+    α, β = -r[1], one(eltype(r))
+    for k in 1:n-1
+        β *= (1-α^2)
+        r_k, y_k = @views r[1:k], y[1:k]
+        α = -(r[k+1] + reverse_dot(r_k, y_k)) / β
+        reverse_increment!(y_k, y_k, α)
+        y[k+1] = α
+    end
+    return y
+end
+
+"""
+```
+    trench(T::SymmetricToeplitz)
+```
+Trench's algorithm (see page 213 of Golub & van Loan) computes inverse of a
+symmetric positive definite Toeplitz matrix in `O(n^2)` operations.
+"""
+function trench(T::SymmetricToeplitz)
+    n = checksquare(T)
+    B = zeros(eltype(T), n, n)
+    trench!(B, T)
+end
+
+"""
+```
+    trench!(B::AbstractMatrix, T::SymmetricToeplitz)
+```
+Same as `trench` but uses `B` to store the upper triangular portion of the inverse
+of T and returns a `Symmetric` wrapper of `B`.
+"""
+function trench!(B::AbstractMatrix, T::SymmetricToeplitz)
+    r_0 = T.vc[1]
+    r = @view T.vc[2:end]
+    if !isone(r_0) # levinson implementation assumes normalization of diagonal
+        r /= r_0 # not in place so that we don't mutate T
+    end
+    S = trench!(B, r)
+    if !isone(r_0)
+        @. B /= r_0
+    end
+    return S
+end
+
+# computes inverse of K = SymmetricToeplitz(vcat(1, r))
+function trench(r::AbstractVector)
+    n = length(r) + 1
+    B = zeros(eltype(r), n, n)
+    trench!(B, r)
+end
+
+# computes inverse of K = SymmetricToeplitz(vcat(1, r))
+# uses B to store the inverse
+# NOTE: only populates entries on upper triangle since the inverse is symmetric
+function trench!(B::AbstractMatrix, r::AbstractVector)
+    n = checksquare(B)
```

Comment:
```suggestion
    n = LinearAlgebra.checksquare(B)
```

When resolving the merge conflict, I might have removed the import of `checksquare`...